### ISSUE TITLE
OrderSearchRequest's Query is "string" typed and causes 400 Bad Request 

### ISSUE THREAD

**Issue by andrewchaa** [commercetools/commercetools-dotnet-core-sdk-v2#138]
**Describe the bug**
As the type of `Query` of `OrderSearchRequest` is string, the query get serialized into string content like the below.

```json
{   
  "query": "{ \"exact\": {\"field\": \"itemShippingAddresses.postalCode\", \"value\": \"EC1R 5EN\"  } }",
  "limit": 20,
  "offset": 0
}
```

**To Reproduce**

```csharp
ApiRoot
  .WithProjectKey("test_projectKey")
  .Orders()
  .Search()
  .Post(new OrderSearchRequest {
    Query = JsonSerializer.Serialize(new {
        exact = new {
          field = "itemShippingAddresses.postalCode",
          value = "EC1R 5EN"
        }
      }
  })
```

This call returns with 400 Bad Request

**Expected behavior**
It should send a valid json as POST request body

```json
{   
  "query": {
     "exact": {
       "field": "itemShippingAddresses.postalCode", 
       "value":  "EC1R 5EN"  
      }
   },
  "limit": 20,
  "offset": 0
}

```

**Screenshots/Code snippet**
If applicable, add screenshots or code samples to help explain your problem.

**Stack information (please complete the following information):**
 - dotnet: Core 3.1
 - SDK: [e.g. v2.2]

**Additional context**

**Comment by jenschude** [comment]
It's a nice catch. I opened a PR in the upstream repository, so that the type can be fixed. As the query body is not yet completely modelled the query field will be an empty interface which means that for a query a class has to be defined.

I'm still not pleased with this. So could be that we have to change this in the future.

**Comment by andrewchaa** [comment]
I agree that the fix on the upstream is more elegant.

**Comment by jenschude** [comment]
Fixed with 2.5.1

**Comment by senj** [comment]
Hi @jenschude , I think the same is still true for the Sort property, also string but array is expected.

**Comment by jenschude** [comment]
Thanks for the report! Will tackle it

**Comment by jenschude** [comment]
Released 5.2.0 with the fix

Thanks for reporting @senj
